docs/src/main/sphinx/security/inspect-jks.rst

```diff
@@ -0,0 +1,135 @@
+
+* The keystore may contain either a private key (``Entry type:
+  PrivateKeyEntry``) or certificate (``Entry type: trustedCertEntry``) or both.
+*  Modern browsers now enforce 398 days as the maximum validity period for a
+   certificate. Look for the ``Valid from ... until`` entry, and make sure the
+   time span does not exceed 398 days.
+*  Modern browsers and clients require the **SubjectAlternativeName** (SAN)
+   field. Make sure this shows the DNS name of your server, such as
+   ``DNS:clustercoord.example.com``. Certificates without SANs are not
+   supported.
+
+  Example:
+
+  .. code-block:: text
+
+    SubjectAlternativeName [
+        DNSName:  cluster.example.com
+    ]
+
+If your keystore shows valid information for your cluster, proceed to configure
+the Trino server, as described in :ref:`cert-placement` and
+:ref:`configure-https`.
+
+The rest of this page describes additional steps that may apply in certain
+circumstances.
+
+.. _import_to_keystore:
+
+Extra: add PEM to keystore
+--------------------------
+
+Your site may have standardized on using JKS semantics for all servers. If a
+vendor sends you a PEM-encoded certificate file for your Trino server, you can
+import it into a keystore with a command like the following. Consult ``keytool``
+references for different options.
+
+.. code-block:: shell
+
+  keytool -trustcacerts -import -alias cluster -file localhost.pem -keystore localkeys.jks
+
+If the specified keystore file exists, ``keytool`` prompts for its password. If
+you are creating a new keystore, ``keytool`` prompts for a new password, then
+prompts you to confirm the same password. ``keytool`` shows you the
+contents of the key being added, similar to the ``keytool -list`` format, then
+prompts:
+
+.. code-block:: text
+
+  Trust this certificate? [no]:
+
+Type ``yes`` to add the PEM certificate to the keystore.
+
+The ``alias`` name is an arbitrary string used as a handle for the certificate
+you are adding. A keystore can contain multiple keys and certs, so ``keytool``
+uses the alias to address individual entries.
+
+.. _cli_java_truststore:
+
+Extra: Java truststores
+-----------------------
+
+.. note::
+
+    Remember that there may be no need to identify a local truststore when
+    directly using a signed PEM-encoded certificate, independent of a keystore.
+    PEM certs can contain the server's private key and the certificate chain all
+    the way back to a recognzied CA.
+
+Truststore files contain a list of :ref:`Certificate Authorities <glossCA>`
+trusted by Java to validate the private keys of servers, plus a list of the
+certificates of trusted TLS servers. The standard Java-provided truststore file,
+``cacerts``, is part of your Java installation in a standard location.
+
+Keystores normally rely on the default location of the system truststore, which
+therefore does not need to be configured.
+
+However, there are cases in which you need to append to the standard truststore
```

Comment:
OK, the whole truststore section is redone and smaller. Instead of using your four point sequence, I punted to "contact your network admins" for all client config.